libindy/src/services/pool/events.rs

```diff
@@ -203,8 +205,13 @@ impl Into<Option<RequestEvent>> for PoolEvent {
                 _parse_msg(&msg).map(|parsed|
                     match parsed {
                         //TODO change mapping for CatchupReq. May be return None
-                        //TODO: REMOVE UNWRAP!!!!!
-                        Message::CatchupReq(_) => RequestEvent::CatchupReq(MerkleTree::from_vec(Vec::new()).unwrap(), 0, vec![]),
+                        Message::CatchupReq(_) => RequestEvent::CatchupReq(
+                            MerkleTree {
+                                root: Tree::empty(EMPTY_HASH_BYTES.to_vec()),
```

Comment:
Shouldn't we use `MerkleTree::default()` here as well?

Reply:
The enum type Tree is different to the type MerkleTree. 
Tree::empty is only used once, so I think implementing default is not needed.